### PR TITLE
Update ReadMe

### PR DESCRIPTION
**Pull request by desistefanova**
Added links to ReadMe.
Add more information about device sync

### CODE REVIEW

**Comment by desistefanova** [comment]
@mongodben I will appreciate it if you can review this ReamMe.

File: README.md
Comment (nielsenko) [inline README.md:328]:
```suggestion
* Create an account on [cloud.mongodb.com](https://cloud.mongodb.com) - follow the instructions: [Get Started with Atlas](https://www.mongodb.com/docs/atlas/getting-started) or [Get Started with a template app](https://www.mongodb.com/docs/atlas/app-services/tutorial/flutter/)
```

**Review by nielsenko — APPROVED** [review]

File: README.md
Comment (mongodben) [inline README.md:91]:
```suggestion
Use [realm](https://pub.dev/packages/realm) package for Flutter and [realm_dart](https://pub.dev/packages/realm_dart) package for Dart standalone applications.
```

File: README.md
Comment (mongodben) [inline README.md:93]:
```suggestion
For complete documentation of the SDKs, go to the [Realm SDK documentation](https://docs.mongodb.com/realm/sdk/flutter/).
```

File: README.md
Comment (mongodben) [inline README.md:95]:
```suggestion
If you are using the Realm SDK for the first time, refer to the [Quick Start documentation](https://www.mongodb.com/docs/realm/sdk/flutter/quick-start/).
```

File: README.md
Comment (mongodben) [inline README.md:97]:
```suggestion
To learn more about using Realm with Atlas App Services and Device Sync, refer to the following Realm SDK documentation:

- [App Services Overview](https://www.mongodb.com/docs/realm/sdk/flutter/app-services/)
- [Device Sync Overview](https://www.mongodb.com/docs/realm/sdk/flutter/sync/)
```

File: README.md
Comment (mongodben) [inline README.md:101]:
```suggestion
* Realm Database data types Set and Map are not supported yet.
```

File: README.md
Comment (mongodben) [inline README.md:102]: suggestion: might also want to add other unsupported features.

for example, on the app Services side, MongoDB Data Access isn't supported. 

not sure what else is missing versus what a Realm SDK _should_ have, but here'd be a good place to mention.

File: README.md
Comment (mongodben) [inline README.md:109]:
```suggestion
* The Realm Flutter SDK supports the platforms iOS, Android, Windows, MacOS, and Linux. The Dart Standalone SDK supports Windows, MacOS and Linux.
```

File: README.md
Comment (mongodben) [inline README.md:112]:
```suggestion
* For Flutter Desktop environment setup, see [Desktop support for Flutter](https://docs.flutter.dev/desktop) in the Flutter documentation.
```

File: README.md
Comment (mongodben) [reply README.md:109]: note that removed part about Dart Standalone b/c discussed in following section

File: README.md
Comment (mongodben) [inline README.md:279]:
```suggestion
# Realm Dart Standalone SDK
```

File: README.md
Comment (mongodben) [inline README.md:105]:
```suggestion
The Realm Flutter SDK's package name is [realm](https://pub.dev/packages/realm).
```

File: README.md
Comment (mongodben) [inline README.md:281]:
```suggestion
The Realm Dart Standalone SDK package is [realm_dart](https://pub.dev/packages/realm_dart).
```

File: README.md
Comment (mongodben) [inline README.md:320]:
```suggestion
* For more information on using the Realm Dart Standalone SDK see the Realm Flutter SDK usage above.
```

File: README.md
Comment (mongodben) [inline README.md:322]:
```suggestion
# Device Sync with Realm Flutter and Dart Standalone SDKs
```

File: README.md
Comment (desistefanova) [inline README.md:287]: To be aligned to [Update to analyzer 5+](https://github.com/realm/realm-dart/pull/1072/)

File: README.md
Comment (mongodben) [inline README.md:326]:
```suggestion
# Set up Atlas App Services
```

File: README.md
Comment (mongodben) [reply README.md:328]:
```suggestion
* Create an account on [MongoDB Atlas](https://cloud.mongodb.com).
```

File: README.md
Comment (mongodben) [inline README.md:332]:
```suggestion
1. Read [Authentication Providers](https://www.mongodb.com/docs/atlas/app-services/authentication/providers/) to see how to configure the appropriate authentication provider type.
1. Go to the **Device Sync** menu and [Enable Flexible Sync](https://www.mongodb.com/docs/atlas/app-services/sync/configure/enable-sync/#enable-flexible-sync).
1. [Find and Copy the App ID](https://www.mongodb.com/docs/atlas/app-services/reference/find-your-project-or-app-id/) of your new application.
```

File: README.md
Comment (mongodben) [inline README.md:334]:
```suggestion
## Use Device Sync in the Realm SDK
```

File: README.md
Comment (mongodben) [inline README.md:369]:
````suggestion
1. Set Atlas App ID as a constant.

   ``` dart
   String appId = "tasktracker-fleld";
   ```

1. Initialize the App Services `App` client and authenticate a user.

   ``` dart
   final appConfig = AppConfiguration(appId);
   final app = App(appConfig);
   final user = await app.logIn(Credentials.anonymous());
   ```
1. Open a synced realm.
   ``` dart
   final flxConfig = Configuration.flexibleSync(user, [Task.schema]);
   final realm = Realm(flxConfig);
   ```

1. Add a sync subscription and write data.
   
   ``` dart
   realm.subscriptions.update((mutableSubscriptions) {
   mutableSubscriptions.add(realm.query<Task>(r'status == $0 AND progressMinutes == $1', ["completed", 100]));
   });
   realm.write(() {
     realm.add(Task(ObjectId(), "Send an email", "completed", 4));
     realm.add(Task(ObjectId(), "Create a meeting", "completed", 100));
     realm.add(Task(ObjectId(), "Call the manager", "init", 2));
   });

   await realm.subscriptions.waitForSynchronization();
   realm.close();
   ```

   Only data matching the query in the subscription will be synced to the server and only data matching the subscription will be downloaded to the local device realm file.
````

File: README.md
Comment (mongodben) [inline README.md:371]:
```suggestion
To learn more about how to sync Realm using Device Sync, refer to the [Quick Start with Sync documentation](https://www.mongodb.com/docs/realm/sdk/flutter/quick-start/#sync-realm-with-mongodb-atlas).
```

**Review by mongodben — COMMENTED** [review]
nice improvements! cosmetic changes throughout.

File: README.md
Comment (mongodben) [reply README.md:335]: by doing this, it renders as a numbered list. you could also do `1. `, `2. `, etc.

Comment (mongodben) [reply README.md:335]: https://stackoverflow.com/questions/64918160/markdown-how-to-make-numbered-list-without-writing-the-actual-numbers

File: README.md
Comment (blagoev) [inline README.md:114]: In general we should not be mentioning minor versions unless needed as the case with the Dart bug. Also in master we merged a PR which allows `^3.0.0`

File: README.md
Comment (blagoev) [inline README.md:115]:
```suggestion
* For Flutter Desktop environment setup, see [Desktop support for Flutter](https://docs.flutter.dev/desktop).
```

File: README.md
Comment (blagoev) [reply README.md:290]: I don't understand the comment. You want to block this PR until the other PR lands ?

File: README.md
Comment (blagoev) [inline README.md:323]:
```suggestion
* For more information on using the Realm Dart Standalone see the Realm Flutter usage above.
```
as mentioned above we don't need to emphasize the word SDK each time

File: README.md
Comment (blagoev) [reply README.md:335]: nice tip. Still if we go with a list here it would be better imo if we use the 1,2,3 since its more readable while editing the file as well.

File: README.md
Comment (blagoev) [inline README.md:325]:
```suggestion
# Device Sync with Realm Flutter and Dart SDKs 
```
The Dart Standalone goes with many names so we could skip mentioning it too much. Not sure which one is the most popular or if there is one  so I suggest we just drop it here.

File: README.md
Comment (blagoev) [inline README.md:327]: What does it mean to `Run the Realm`? I think we need to rephrase it here not sure how since it is not clear to me what is the purpose of this sentence? Provide the links to the docs?

File: README.md
Comment (blagoev) [inline README.md:332]: this looks mismatched. We start with a bullet and then switch to numbered list.

Comment (blagoev) [inline README.md:332]: I suggest we pick one thing here and guide people to use that one. So perhaps the template app here only

File: README.md
Comment (blagoev) [inline README.md:329]: Should we change to numbered list here. Is this a precondition to `Use Device Sync in the Realm SDK`

File: README.md
Comment (blagoev) [inline README.md:342]: I suggest we provide longer example here. so combining this example with the next one will be more clear.

File: README.md
Comment (blagoev) [inline README.md:339]: I think these should be part of a numbered list

File: README.md
Comment (blagoev) [inline README.md:354]:
```suggestion
  final config = Configuration.flexibleSync(user, [Task.schema]);
```

File: README.md
Comment (blagoev) [inline README.md:369]: why `waitForSynchronization` after we did a `realm write`

Comment (blagoev) [reply README.md:369]: shouldn't this be before the write

File: README.md
Comment (blagoev) [inline README.md:374]:
```suggestion
To learn more about how to sync data with Realm using Device Sync, refer to the [Quick Start with Sync documentation](https://www.mongodb.com/docs/realm/sdk/flutter/quick-start/#sync-realm-with-mongodb-atlas).
```

File: README.md
Comment (blagoev) [inline README.md:325]:
```suggestion
# Sync data with Realm Flutter and Dart using Device Sync
```

**Review by blagoev — CHANGES_REQUESTED** [review]
added some suggestions

File: README.md
Comment (desistefanova) [reply README.md:290]: I don't want to block this PR. I've just added a marker in case the other PR goes first.

File: README.md
Comment (desistefanova) [reply README.md:105]: We have this only: https://www.mongodb.com/docs/realm/sdk/flutter/#release-candidate-limitations.

File: README.md
Comment (desistefanova) [reply README.md:114]: Yes, probably I will have to fix it in the main branch to be also 3.0.3.

File: README.md
Comment (desistefanova) [reply README.md:332]: Fixed.

Comment (desistefanova) [reply README.md:332]: OK, I choose `Create an App with Atlas App Services UI` because this is the exact instruction only for creating an App.

File: README.md
Comment (desistefanova) [reply README.md:323]: OK

File: README.md
Comment (desistefanova) [reply README.md:325]: OK

Comment (desistefanova) [reply README.md:325]: Replaced with "Sync data with Realm Flutter and Dart using Device Sync"

File: README.md
Comment (desistefanova) [reply README.md:376]: Done

File: README.md
Comment (desistefanova) [reply README.md]: I moved it above.

File: README.md
Comment (desistefanova) [reply README.md]: Done

File: README.md
Comment (desistefanova) [reply README.md]: Edited.

File: README.md
Comment (desistefanova) [reply README.md]: Moved to the next part of code.

File: README.md
Comment (desistefanova) [reply README.md:329]: I don't understand what should be numbered list. The title or everything after it?

File: README.md
Comment (desistefanova) [reply README.md:114]: Added PR https://github.com/realm/realm-dart/pull/1074

File: README.md
Comment (desistefanova) [reply README.md:290]: Sorry, @blagoev, I deleted your comment by mistake. It was duplicated somehow.
The comment was: "I don't understand the comment. You want to block this PR until the other PR lands ?"

File: README.md
Comment (desistefanova) [reply README.md:105]: @blagoev commented: "Do we have such limitations page in the docs? We could link to that from here instead?"

Comment (desistefanova) [reply README.md:105]: We have this only: https://www.mongodb.com/docs/realm/sdk/flutter/#release-candidate-limitations. But it doesn't list what is still not implemented. Maybe I have to delete this section. What do you think?

Comment (mongodben) [reply README.md:105]: i think we can delete this section, then. 

There's no official list of features that a Realm SDK 'should' have, as far as I know. 

Plus, this will be annoying to update as features get added.

File: README.md
Comment (mongodben) [inline README.md:323]:
```suggestion
* For more information on using the Realm Dart Standalone SDK see the Realm Flutter SDK usage above.
```

File: README.md
Comment (mongodben) [inline README.md:331]:
```suggestion
1. Create an account on [cloud.mongodb.com](https://cloud.mongodb.com). Follow the instructions: [Register a new Atlas Account](https://www.mongodb.com/docs/atlas/tutorial/create-atlas-account/#register-a-new-service-account).
```

File: README.md
Comment (mongodben) [inline README.md:346]:
```suggestion
   ```

```

File: README.md
Comment (mongodben) [inline README.md:348]:
```suggestion

   ``` dart
```

File: README.md
Comment (mongodben) [inline README.md:368]:
```suggestion
   
   Only data matching the query in the subscription will be synced to the server and only data matching the subscription will be downloaded to the local device realm file.

   ``` dart
   realm.subscriptions.update((mutableSubscriptions) {
   mutableSubscriptions.add(realm.query<Task>(r'status == $0 AND progressMinutes == $1', ["completed", 100]));
   });
   await realm.subscriptions.waitForSynchronization();
   realm.write(() {
     realm.add(Task(ObjectId(), "Send an email", "completed", 4));
     realm.add(Task(ObjectId(), "Create a meeting", "completed", 100));
     realm.add(Task(ObjectId(), "Call the manager", "init", 2));
   });
   realm.close();
   ```
```

Comment (mongodben) [reply README.md:368]: moved description before code example

**Review by mongodben — APPROVED** [review]
couple more small suggestions throughout and i recommend removing the `## Limitations` section. once that's addressed, lgtm!

File: README.md
Comment (desistefanova) [reply README.md]: Changed to "The usage of the Realm Dart SDK is the same like the Realm Flutter usage above."

File: README.md
Comment (desistefanova) [reply README.md]: Done

File: README.md
Comment (blagoev) [inline README.md:104]: We mention this in the beginning. Should we have this here as well?

File: README.md
Comment (blagoev) [inline README.md:108]:
```suggestion
* Realm Flutter supports the platforms iOS, Android, Windows, MacOS and Linux.
```

File: README.md
Comment (blagoev) [inline README.md:284]:
```suggestion
* Realm Dart supports the platforms Windows, Mac and Linux.
```

File: README.md
Comment (blagoev) [inline README.md:319]:
```suggestion
* The usage of the Realm Dart SDK is the same like the Realm Flutter above.
```

**Review by blagoev — APPROVED** [review]
some small final comments

File: README.md
Comment (desistefanova) [reply README.md:104]: It is good to have the link here in the relevant section. But you are right that we already mention the names above. Is it OK, to change the text like this:
"Realm Flutter package is published to [realm](https://pub.dev/packages/realm)."?
The same for Dart.

**Review by blagoev — APPROVED** [review]

**Pull request merged**
